extensions/mssql/src/features.ts

```diff
@@ -43,7 +44,17 @@ export class AccountFeature implements StaticFeature {
 		let timeToLiveInSeconds = 10;
 		this.tokenCache = new DataItemCache(this.getToken, timeToLiveInSeconds);
 		this._client.onRequest(contracts.SecurityTokenRequest.type, async (request): Promise<contracts.RequestSecurityTokenResponse | undefined> => {
-			return this.tokenCache.getData(request);
+			return await this.tokenCache.getData(request);
+		});
+		this._client.onNotification(contracts.RefreshTokenNotification.type, async (request) => {
+			// Refresh token, then inform client the token has been updated. This is done as separate notification messages due to the synchronous processing nature of STS currently https://github.com/microsoft/azuredatastudio/issues/17179
+			let result = await this.refreshToken(request);
+			if (!result) {
+				void window.showErrorMessage(localizedConstants.tokenRefreshFailed('intellisense'));
```

Comment:
1. We should avoid using the term "Intellisense" in VS Code. That's specifically a VS thing. Use `autocompletion` instead
2. We shouldn't be doing this here - theoretically anything could call this notification and so we should have the request itself contain the context string to insert. Or even better I would suggest having the request have an optional error message to display - and if that isn't defined then don't show anything at all so callers can control whether this is surfaced to the user.

Reply:
I'm not sure I understand the second point, why can't we just have it in ADS?

Reply:
In the interest of getting the user unblocked, can this be addressed in a separate PR?

Reply:
It shouldn't be a huge task (and your STS PR is still open anyways so seems easy enough to add in there) - but if you really don't think you have time to address this now I guess it's not super critical...